src/groupByKey.ts

```diff
@@ -2,6 +2,18 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: Array<T>, key: keyof T): GroupsMap<T> {
+  const grouppedData: GroupsMap<T> = {};
+
+  items.forEach((item) => {
+    const keyToGroupBy = String(item[key]);
```

Comment:
it's a good solution for this task, but it would be great to restrict not primitive types to be value of `T` object. for example, `String({ key: "value" })` will be `'[object Object]'`